src/bio/bio_buffer.c

```diff
@@ -1510,7 +1510,7 @@ bio_copy_prep(struct bio_io_context *ioctxt, struct bio_sglist *bsgl_src,
 	copy_desc->bcd_iod_dst->bd_copy_dst = 1;
 	rc = bio_iod_prep(copy_desc->bcd_iod_dst, BIO_CHK_TYPE_LOCAL, NULL, 0);
 	if (rc) {
-		rc = bio_iod_post(copy_desc->bcd_iod_src);
+		rc = bio_iod_post(copy_desc->bcd_iod_src, 0);
```

Comment:
should it pass rc for bio_iod_post() to skip the dma_rw()? (bio_iod_post(copy_desc->bcd_iod_src, rc)).

Reply:
For a copy operation, the 'rc' (for data copy) should only affect the bio_iod_post() for copy target (the bio IOD for update), copy source is for fetching, and it's already done before the data copy (in bio_iod_prep() for source IOD), so the data copy error code actually doesn't affect bio_iod_post() for copy source.